move/sources/hongbao.move

```diff
@@ -135,6 +145,21 @@ module addr::hongbao {
         }
     }
 
+    struct Config has key {
+        paused: bool
+    }
+
+    /// Pause claiming and creating gifts
+    public entry fun set_paused(caller: &signer, pause: bool) acquires Config {
+        if (exists<Config>(@addr)) {
+            let config = borrow_global_mut<Config>(@addr);
+            config.paused = pause;
+        } else {
+            let config = Config { paused: pause };
+            move_to(caller, config);
+        }
+    }
```

Comment:
the `else` here isn't what you want, unless you also add an assert that `address_of(caller) == @addr`

Reply:
ah yes forgot to add that. i had a thought that if people wanted to put a struct on their account they could tho without consequences